### PR TITLE
sqlite3: Add test case with database path contains whitespace

### PR DESCRIPTION
**Pull request by ganboonhong**
Fix https://github.com/golang-migrate/migrate/issues/499

Since URL will be encoded using percent-encoding in
https://github.com/golang-migrate/migrate/blob/550193e3e0b3d8f4858a625ed9b468751cdeb620/database/sqlite3/sqlite3.go#L95

`url` like `/path/to database/` (it contains whitespace) will be encoded to `/path/to%20database/`, which causes

https://github.com/golang-migrate/migrate/blob/550193e3e0b3d8f4858a625ed9b468751cdeb620/database/sqlite3/sqlite3.go#L47

to return an error.

### CODE REVIEW

File: database/sqlite3/sqlite3.go [hunk 1]
```diff
@@ -97,6 +97,11 @@ func (m *Sqlite) Open(url string) (database.Driver, error) {
 		return nil, err
 	}
 	dbfile := strings.Replace(migrate.FilterCustomQuery(purl).String(), "sqlite3://", "", 1)
+	dbfile, err = nurl.QueryUnescape(dbfile)
```
Comment (dhui) [inline database/sqlite3/sqlite3.go:100]: The whole connection string cannot be unescaped as [the options need to be escaped](https://github.com/mattn/go-sqlite3#connection-string).
I think we'll need to reconstruct the connection string w/ the host & path unescaped unless you can think of a cleaner way...

[Example](https://play.golang.org/p/4h83CEXdyep) of what the parsed connection string looks like

File: database/sqlite3/sqlite3_test.go [hunk 2]
```diff
@@ -160,3 +160,23 @@ func TestNoTxWrapInvalidValue(t *testing.T) {
 		assert.Contains(t, err.Error(), "invalid syntax")
 	}
 }
+
+func TestMigrateWithDirectoryNameContainsWhitespaces(t *testing.T) {
```
Comment (dhui) [inline database/sqlite3/sqlite3_test.go:164]: Add a test to ensure options specified in the connection string remain escaped

**Review by dhui — CHANGES_REQUESTED** [review]
Thanks for the PR and tests

File: database/sqlite3/sqlite3_test.go [hunk 3]
```diff
@@ -160,3 +160,23 @@ func TestNoTxWrapInvalidValue(t *testing.T) {
...
+		return
+	}
+	defer func() {
+		if err := os.RemoveAll(dir); err != nil {
+			t.Error(err)
+		}
+	}()
+	t.Logf("DB path : %s\n", filepath.Join(dir, "sqlite3.db"))
+	p := &Sqlite{}
+	addr := fmt.Sprintf("sqlite3://%s%s", "file:", filepath.Join(dir, "sqlite3.db"))
```
Comment (ganboonhong) [inline database/sqlite3/sqlite3_test.go:176]: According to SQLite [document](https://www.sqlite.org/uri.html):
> SQLite uses the "file:" URI syntax to identify database files. SQLite strives to interpret file: URIs in exactly the same way as popular web-browsers



I found that by prefixing `file:` to the DSN, then SQLite driver will interpret URI correctly. So unescaping the connection string is no longer needed. This test could also be treated as a reference on how to handle directory name with whitespaces.

Comment (dhui) [reply database/sqlite3/sqlite3_test.go:176]: Can you clean up this a bit? e.g.
```golang
dbPath := filepath.Join(dir, "sqlite3.db")
t.Log("DB path:", dbPath)
addr := fmt.Sprintf("sqlite3://file:%s", dbPath)
```

**Review by dhui — CHANGES_REQUESTED** [review]
Oh nice find with the `file:` prefix!

I agree with you and think there's still value in testing opening paths with spaces in them.

**Comment by ganboonhong** [comment]
Thanks for the suggestion! Already modified the code.

File: database/sqlite3/sqlite3_test.go (hunk 3 above)
Comment (ganboonhong) [reply database/sqlite3/sqlite3_test.go]: Thanks for the suggestion! Already modified the code.

**Comment by ganboonhong** [comment]
Thanks for the suggestion! Already modified the code.

**Review by dhui — APPROVED** [review]
